student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java

```diff
@@ -0,0 +1,94 @@
+package tds.student.services.remote;
+
+import TDS.Shared.Exceptions.ReturnStatusException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import tds.exam.ApproveAccommodationsRequest;
+import tds.student.services.abstractions.IAccommodationsService;
+import tds.student.sql.abstractions.ExamRepository;
+import tds.student.sql.data.Accommodations;
+import tds.student.sql.data.OpportunityInstance;
+
+@Service("integrationAccommodationsService")
+@Scope("prototype")
+public class RemoteAccommodationsService implements IAccommodationsService {
+  private final IAccommodationsService legacyAccommodationsService;
+  private final boolean isRemoteExamCallsEnabled;
+  private final boolean isLegacyCallsEnabled;
+  private final ExamRepository examRepository;
+
+  @Autowired
+  public RemoteAccommodationsService(
+    @Qualifier("legacyAccommodationsService") IAccommodationsService legacyAccommodationsService,
+    @Value("${tds.exam.remote.enabled}") Boolean remoteExamCallsEnabled,
+    @Value("${tds.exam.legacy.enabled}") Boolean legacyCallsEnabled,
+    ExamRepository examRepository) {
+
+    if (!remoteExamCallsEnabled && !legacyCallsEnabled) {
+      throw new IllegalStateException("Remote and legacy calls are both disabled.  Please check progman configuration");
+    }
+
+    this.isRemoteExamCallsEnabled = remoteExamCallsEnabled;
+    this.legacyAccommodationsService = legacyAccommodationsService;
+    this.isLegacyCallsEnabled = legacyCallsEnabled;
+    this.examRepository = examRepository;
+  }
+
+  @Override
+  public List<Accommodations> getTestee(String testKey, boolean isGuestSession, long testeeKey) throws ReturnStatusException {
+    return legacyAccommodationsService.getTestee(testKey, isGuestSession, testeeKey);
+  }
+
+  @Override
+  public void approve(OpportunityInstance oppInstance, List<String> segmentsAccommodationData) throws ReturnStatusException {
+
+    if (isLegacyCallsEnabled) {
+      legacyAccommodationsService.approve(oppInstance, segmentsAccommodationData);
+    }
+
+    if (!isRemoteExamCallsEnabled || segmentsAccommodationData == null) {
```

Comment:
The legacy code also bails out early if segmentData == null;  (line 275,275 in AccommodationService.java) Ultimately I guess we can just bail out immediately if segmentData == null, and not call either service. Strange that it doesnt result in any sort of warning or error.